File: myWork/app/src/main/java/com/shindefirm/shopapp/modal/PartyDataModel.java
package com.shindefirm.shopapp.modal;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class PartyDataModel implements Parcelable {
    //    "Id": 1,
//            "PartyName": "Nationalist Congress Party",
//            "PartyShortCode": "NCP"
    @SerializedName("Id")
    private String id;
    @SerializedName("PartyName")
    private String partyName;
    @SerializedName("PartyShortCode")
    private String partyShortCode;

    public PartyDataModel() {
        this.id = "";
        this.partyName = "";
        this.partyShortCode = "";
    }

    public PartyDataModel(String id, String partyName, String partyShortCode) {
        this.id = id;
        this.partyName = partyName;
        this.partyShortCode = partyShortCode;
    }

    protected PartyDataModel(Parcel in) {
        id = in.readString();
        partyName = in.readString();
        partyShortCode = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(id);
        dest.writeString(partyName);
        dest.writeString(partyShortCode);
    }

    @Override
    public int describeContents() {
        return hashCode();
    }

    public static final Creator<PartyDataModel> CREATOR = new Creator<PartyDataModel>() {
        @Override
        public PartyDataModel createFromParcel(Parcel in) {
            return new PartyDataModel(in);
        }

        @Override
        public PartyDataModel[] newArray(int size) {
            return new PartyDataModel[size];
        }
    };

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getPartyName() {
        return partyName;
    }

    public void setPartyName(String partyName) {
        this.partyName = partyName;
    }

    public String getPartyShortCode() {
        return partyShortCode;
    }

    public void setPartyShortCode(String partyShortCode) {
        this.partyShortCode = partyShortCode;
    }

    @Override
    public String toString() {
        return partyName;
    }
}

File: myWork/app/src/main/java/com/shindefirm/shopapp/modal/PartyDataModel.kt
package com.shindefirm.shopapp.modal

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import android.os.Parcel
import android.os.Parcelable.Creator
import com.shindefirm.shopapp.modal.PartyDataModel

class PartyDataModel : Parcelable {
    //    "Id": 1,
    //            "PartyName": "Nationalist Congress Party",
    //            "PartyShortCode": "NCP"
    @SerializedName("Id")
    var id: String?

    @SerializedName("PartyName")
    var partyName: String

    @SerializedName("PartyShortCode")
    var partyShortCode: String?

    constructor() {
        id = ""
        partyName = ""
        partyShortCode = ""
    }

    constructor(id: String?, partyName: String, partyShortCode: String?) {
        this.id = id
        this.partyName = partyName
        this.partyShortCode = partyShortCode
    }

    protected constructor(`in`: Parcel) {
        id = `in`.readString()
        partyName = `in`.readString()!!
        partyShortCode = `in`.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(id)
        dest.writeString(partyName)
        dest.writeString(partyShortCode)
    }

    override fun describeContents(): Int {
        return hashCode()
    }

    override fun toString(): String {
        return partyName
    }

    companion object {
        @JvmField
        val CREATOR: Creator<PartyDataModel?> = object : Creator<PartyDataModel?> {
            override fun createFromParcel(`in`: Parcel): PartyDataModel? {
                return PartyDataModel(`in`)
            }

            override fun newArray(size: Int): Array<PartyDataModel?> {
                return arrayOfNulls(size)
            }
        }
    }
}